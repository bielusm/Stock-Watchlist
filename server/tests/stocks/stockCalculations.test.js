const { getLast52Data } = require('../../stocks/stockCalculations');
const ibmData = require('../fixtures/stocks/ibmStocks.json');

describe('stockCalculation tests', () => {
  describe('getLast52Data', () => {
    test('should get last 52 datapoints', () => {
      const last52 = getLast52Data(ibmData['Weekly Time Series']);
      expect(last52).toHaveLength(52);
      expect(last52).toMatchSnapshot();
    });
  });
});
