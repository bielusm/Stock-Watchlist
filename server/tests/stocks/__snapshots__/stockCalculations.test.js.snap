// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stockCalculation tests getLast52Data should get last 52 datapoints 1`] = `
Array [
  Object {
    "date": "2020-03-27",
    "high": "113.1500",
  },
  Object {
    "date": "2020-03-20",
    "high": "107.4100",
  },
  Object {
    "date": "2020-03-13",
    "high": "124.8800",
  },
  Object {
    "date": "2020-03-06",
    "high": "136.0950",
  },
  Object {
    "date": "2020-02-28",
    "high": "148.0500",
  },
  Object {
    "date": "2020-02-21",
    "high": "151.8900",
  },
  Object {
    "date": "2020-02-14",
    "high": "155.6000",
  },
  Object {
    "date": "2020-02-07",
    "high": "158.7500",
  },
  Object {
    "date": "2020-01-31",
    "high": "144.0500",
  },
  Object {
    "date": "2020-01-24",
    "high": "145.7900",
  },
  Object {
    "date": "2020-01-17",
    "high": "138.3300",
  },
  Object {
    "date": "2020-01-10",
    "high": "137.8700",
  },
  Object {
    "date": "2020-01-03",
    "high": "135.9200",
  },
  Object {
    "date": "2019-12-27",
    "high": "136.1500",
  },
  Object {
    "date": "2019-12-20",
    "high": "136.4200",
  },
  Object {
    "date": "2019-12-13",
    "high": "135.6604",
  },
  Object {
    "date": "2019-12-06",
    "high": "134.5000",
  },
  Object {
    "date": "2019-11-29",
    "high": "136.1500",
  },
  Object {
    "date": "2019-11-22",
    "high": "135.3800",
  },
  Object {
    "date": "2019-11-15",
    "high": "137.2000",
  },
  Object {
    "date": "2019-11-08",
    "high": "139.1350",
  },
  Object {
    "date": "2019-11-01",
    "high": "136.6300",
  },
  Object {
    "date": "2019-10-25",
    "high": "135.9300",
  },
  Object {
    "date": "2019-10-18",
    "high": "143.7200",
  },
  Object {
    "date": "2019-10-11",
    "high": "144.5000",
  },
  Object {
    "date": "2019-10-04",
    "high": "147.3500",
  },
  Object {
    "date": "2019-09-27",
    "high": "145.0900",
  },
  Object {
    "date": "2019-09-20",
    "high": "145.0300",
  },
  Object {
    "date": "2019-09-13",
    "high": "145.4600",
  },
  Object {
    "date": "2019-09-06",
    "high": "141.6950",
  },
  Object {
    "date": "2019-08-30",
    "high": "136.0700",
  },
  Object {
    "date": "2019-08-23",
    "high": "136.3300",
  },
  Object {
    "date": "2019-08-16",
    "high": "136.5700",
  },
  Object {
    "date": "2019-08-09",
    "high": "145.1000",
  },
  Object {
    "date": "2019-08-02",
    "high": "152.9500",
  },
  Object {
    "date": "2019-07-26",
    "high": "151.9400",
  },
  Object {
    "date": "2019-07-19",
    "high": "151.5800",
  },
  Object {
    "date": "2019-07-12",
    "high": "142.9200",
  },
  Object {
    "date": "2019-07-05",
    "high": "141.8200",
  },
  Object {
    "date": "2019-06-28",
    "high": "140.1500",
  },
  Object {
    "date": "2019-06-21",
    "high": "139.5400",
  },
  Object {
    "date": "2019-06-14",
    "high": "136.4600",
  },
  Object {
    "date": "2019-06-07",
    "high": "134.7200",
  },
  Object {
    "date": "2019-05-31",
    "high": "132.6400",
  },
  Object {
    "date": "2019-05-24",
    "high": "137.0600",
  },
  Object {
    "date": "2019-05-17",
    "high": "136.1100",
  },
  Object {
    "date": "2019-05-10",
    "high": "140.6900",
  },
  Object {
    "date": "2019-05-03",
    "high": "141.8100",
  },
  Object {
    "date": "2019-04-26",
    "high": "141.3100",
  },
  Object {
    "date": "2019-04-18",
    "high": "145.3900",
  },
  Object {
    "date": "2019-04-12",
    "high": "144.4400",
  },
  Object {
    "date": "2019-04-05",
    "high": "144.2200",
  },
]
`;
